### ISSUE TITLE
Observer Pattern

### ISSUE THREAD

**Issue by cchandel** [zllangct/ecs#5]
Hi,
  Is there a way to use event subscription to call system functions?
Thanks

**Comment by zllangct** [comment]
I think we should change the mode of thinking, this is [ECS](https://en.wikipedia.org/wiki/Entity_component_system), not OOP.    this rep is a concurrency system with lock-free, all logic drived by "frame", ecs benifit from the the independent "system",  your system can call diffrent function by changing component field, a system shoud not communicate with others directly. you can add a new single-use component with some data what you want to send to the system when you want to publish an event.

**Comment by cchandel** [comment]
Ok, 
   That makes sense.
 
   Could you provide example of how both options would work?

   Thanks.

**Comment by zllangct** [comment]
Good idea, I'm planning to do it, the way to program is very different from what we used before, we need more examples

**Comment by cchandel** [comment]
For Eg,
  Lets say the game has the ability to change its speed - which means at speed 2 every thing becomes 2 times as fast, actor movement, game time etc.

  How would this be handled in your view?

**Comment by zllangct** [comment]
I see, you want to modify system by event，such as the time-scale of system and other system field, i think u r right, it's very useful. I will add this capability in future.

**Comment by cchandel** [comment]
Hi,
   Thank you for the fake_game example and for adding events to the ECS system.
Regards

**Comment by zllangct** [comment]
Hi, this rep's apis are unstable, will be optimized or changed continuously  before release version, Thanks for your proposal.

**Comment by zllangct** [comment]
fake_game_server example is incomplete, more usages will be added to this example.

**Comment by cchandel** [comment]
Sure. Understood.
